src/logger.h

```diff
@@ -1,57 +1,23 @@
 #pragma once
 
+#include "URI.h"
 #include <fmt/format.h>
+#include <spdlog/sinks/basic_file_sink.h>
+#include <spdlog/sinks/stdout_color_sinks.h>
+#include <spdlog/spdlog.h>
 #include <string>
-
+//#include <spdlog/sinks/graylog_sink.h>
 #ifdef _MSC_VER
 
+// The levels used in the LOG macro are defined in the spdlog::level namespace
+// in spdlog.h
 #define LOG(level, fmt, ...)                                                   \
-  { dwlog(level, fmt, __FILE__, __LINE__, __FUNCSIG__, __VA_ARGS__); }
-
+  { spdlog::get("filewriterlogger")->log(level, fmt, __VA_ARGS__); }
```

Comment:
This looks up the logger each time we log, which is not great in places where we log very frequently.
Instead, we should get the logger in the constructor of each class which needs to log and remove this macro entirely.